models/retrievers/dense.py

```diff
@@ -48,7 +48,8 @@ def __call__(self, query_or_doc, kwargs):
             }
 
     def collate_fn(self, batch, query_or_doc=None):
-        content = [sample['content'] for sample in batch]
+        key = 'generated_query' if query_or_doc=="query" else "content"
```

Comment:
So the main idea is that after query generation, there is a new field now, "generated_query", which should be used in retrieval instead of "content". (in the default setting they are the same, i.e. query is copied from content)

i changed it in all retrievers which were available at the moment of my PR (seem to be the same in the current version of main branch)

Reply:
So after discussion in Slack with Thibault we concluded it's all good. My change affects only "collate" function and his change with asymetric encoding affects only "__call__" function